### DISCUSSION TITLE
Get multiple string from the Search Box

### DISCUSSION THREAD

**Discussion by dignus93** [microsoft-search/pnp-modern-search#3631] [Q&A]
Hello there!

### **Summary:**
I would like to know if it is possible to get a collection of strings when the user inputs his query in the Search Box. 
Later I want to use this collection to perform an advanced query in the query template of the PnP Search Results

### **Scenario:**
I am using the webpart to filter on a SharePoint list called AnimalList.
In the list, there is a column "Animal". The column is a choice (Cat,Dog,Giraffe,...)
This column is mapped to a managed property called RefinableStringAnimal

The Query Template of the PnP - Search Results is like this:
```
path:"https://mytenant.sharepoint.com/sites/mysitename/Lists/AnimalList" AND
(
	RefinableStringAnimal:{inputQueryText}*
)
```
If the user puts "cat" in the search box you get all the rows with the column Animal equal "cat"
We found the cats, very nice "^.^"

### **Desire:**
If The user puts in the search box "dog cat"
I would like to get all the rows in the list that have the column Animal equal "dog" or equal "cat"

Any idea is welcome! 
Thanks in advance,
Alessandro

**Comment by wobba** [comment]
You need a query modifier to rewrite the input. Which requires coding https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_query_modifications/
